palette/src/yuv/mod.rs

```diff
@@ -0,0 +1,83 @@
+//! YUV types, spaces and standards.
+use float::Float;
+
+use encoding::{TransferFn};
+use rgb::RgbSpace;
+use {Component};
+
+mod quant;
+mod yuv;
+
+/// A YUV standard for analog signal conversion.
+///
+/// In precise terms, YUV identifies an analog encoding of color signal while YCbCr is the digital,
+/// quantized version of that signal.
+pub trait YuvStandard {
+    /// Underlying color space of the RGB signal.
+    type RgbSpace: RgbSpace;
+
+    /// The transfer function from linear RGB space.
+    type TransferFn: TransferFn;
+
+    /// The normalized color difference space.
+    type DifferenceFn: DifferenceFn;
+}
+
+/// Gives the YUV space values of each primary.
+pub trait DifferenceFn {
+    /// The weights of the luminance transform.
+    ///
+    /// The linear transform is assumed to happen after the opto-electric transfer function is
+    /// applied to each color value. This is true for all ITU-R standards. Nevertheless, A
+    /// different form of encoding exists, called YcCbcCbr or constant luminance, which calculates
+    /// the luminance value from the linear RGB values instead to optimize the accuracy of its
+    /// result.
+    ///
+    /// The luminance weights correspond closely to the `Y` components of the `yxy`
+    /// parameterization of the color space primaries. However, they may add up to a value smaller
+    /// than `1` to represent colors appearing brighter than the white point i.e. offer a larger
+    /// dynamic range than otherwise possible.
+    fn luminance<T: Float>() -> [T; 3];
+
+    /// Normalize the difference of luminance and blue channel.
+    fn norm_blue<T: Float>(denorm: T) -> T;
+
+    /// Denormalize the difference of luminance and blue channel.
+    fn denorm_blue<T: Float>(norm: T) -> T;
+
+    /// Normalize the difference of luminance and red channel.
+    fn norm_red<T: Float>(denorm: T) -> T;
+
+    /// Denormalize the difference of luminance and red channel.
+    fn denorm_red<T: Float>(norm: T) -> T;
+}
+
+/// A digital encoding of a YUV color model.
```

Comment:
So this is different from the conversation from other float to int format conversation, right? I see the place where it's used doesn't look like a straight rescale and rounding. Might want to make that clear for the uninitiated, so they know the difference compared to the `into_format` method. 🙂

Reply:
The standard leaves some headroom and footroom, so this is not straightup scaling. I intended to do at least one pure documentation pass and will add some additional context here. Also, code examples overall probably.

Reply:
Ok, that sounds good. :+1: It doesn't have to be a signal processing tutorial, just something that prevents mistakes.